source/cloud/aws/ecs.md

```diff
@@ -1,70 +1,106 @@
 
-b. For Networking, select the default VPC and all the subnets available in that VPC.
+Give the cluster a name e.g.`rapids-cluster`
 
-c. Select "Amazon EC2 instances" for the Infrastructure type and configure your settings: "Linux" operating system, choose an instance type that supports RAPIDS-supported GPUs (see introduction section for list of supported instance types). For this example, we will use `p3.2xlarge`. Finally, select an existing SSH key pair.
+For Networking, select the default VPC and all the subnets available in that VPC
 
-d. All other options can be left at defaults. Review your settings and click on the "Create" button and wait for the cluster creation to complete.
+Select "Amazon EC2 instances" for the Infrastructure type and configure your settings:
 
-**4. Create a Dask cluster:**
+- Operating system: must be Linux-based architecture
+- EC2 instance type: must support RAPIDS-compatible GPUs (Pascal or greater), e.g `p3.2xlarge`
+- Desired capacity: number of maximum instances to launch (default maximum 5)
+- SSH Key pair
+
+Review your settings then click on the "Create" button and wait for the cluster creation to complete.
+
+## Create a Dask cluster
 
 Get the Amazon Resource Name (ARN) for the cluster you just created.
 
-Set `AWS_DEFAULT_REGION` environment variable to your **[default region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions)**:
+Set `AWS_REGION` environment variable to your **[default region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions)**:
 
 ```shell
-$ export AWS_DEFAULT_REGION=[REGION]
+$ export AWS_REGION=[REGION]
 ```
 
-[REGION] = code fo the region being used.
+[REGION] = code for the region being used, e.g. `us-east-1`
 
 Create the ECSCluster object in your Python session:
 
 ```python
 from dask_cloudprovider.aws import ECSCluster
 
 cluster = ECSCluster(
-    cluster_arn=[CLUSTER_ARN], n_workers=[NUM_WORKERS], worker_gpu=[NUM_GPUS]
+    cluster_arn=[CLUSTER_ARN],
+    n_workers=[NUM_WORKERS],
+    worker_gpu=[NUM_GPUS],
+    skip_cleaup=True,
+    execution_role_arn=[EXECUTION_ROLE_ARN],
+    task_role_arn=[TASK_ROLE_ARN],
+    scheduler_timeout="20 minutes",
 )
 ```
 
-[CLUSTER_ARN] = The ARN of an existing ECS cluster to use for launching tasks <br />
-[NUM_WORKERS] = Number of workers to start on cluster creation. <br />
-[NUM_GPUS] = The number of GPUs to expose to the worker, this must be less than or equal to the number of GPUs in the instance type you selected for the ECS cluster (e.g `1` for `p3.2xlarge`).
+````{note}
+When calling this command for the first time, a security group that matches the cluster name will automatically be created. <br />
+
+However, if the cluster creation fails and you want to use the same cluster for subsequent runs of ECSCluster(), you'll need to provide the `security_groups` parameter with the value of the security group, e.g
```

Comment:
If we are suggesting the user creates the security group we should probably mention what rules that security group needs to have.

Reply:
It states that the first time you run this command, you do not need to specify the `"security_group" `param as the `ECSCluster()` command will automatically create one for you ...

But if you need to re-use the same ecs cluster for subsequent runs of `ECSCluster() ,` then you will need to provide the SG that was created prior..

 I hope that makes sense, i have updated it.

Reply:
> But if you need to re-use the same ecs cluster for subsequent runs of ECSCluster() , then you will need to provide the SG that was created prior..

That's interesting. I would expect it to pick up the security group that was created automatically. Perhaps this is a bug in dask-cloudprovier. Anyway thanks for updating here.